pkg/firewall/firewall.go

```diff
@@ -22,25 +22,34 @@ func (nf *noFirewall) Validate(remotePeerPublicKey *ecdsa.PublicKey) error {
 	return nil
 }
 
-// MinimumStakeCachePeriod is the time period the cache maintains the result of
-// the last HasMinimumStake result. We use the cache to minimize calls to
-// Ethereum client.
-const MinimumStakeCachePeriod = 12 * time.Hour
+const (
+	// PositiveMinimumStakeCachePeriod is the time period the cache maintains
+	// the positive result of the last HasMinimumStake check.
+	// We use the cache to minimize calls to Ethereum client.
+	PositiveMinimumStakeCachePeriod = 12 * time.Hour
+
+	// NegativeMinimumStakeCachePeriod is the time period the cache maintains
+	// the negative result of the last HasMinimumStake check.
+	// We use the cache to minimize calls to Ethereum client.
+	NegativeMinimumStakeCachePeriod = 6 * time.Hour
```

Comment:
I'd lower it to one hour, at least initially. Six hours is a lot if someone is not yet very familiar with setting up a node and started it before setting up a delegation.